core/assets/snapshot.go

```diff
@@ -192,3 +198,14 @@ func (s *Service) restorePendingUpdates(_ context.Context, pending *types.Pendin
 
 	return err
 }
+
+func (s *Service) applyMigrations(ctx context.Context, p *types.Asset) {
+	if vgcontext.InProgressUpgradeFrom(ctx, "v0.74.0") {
+		// Prior the introduction of the second bridge, existing assets did not track
+		// the chain ID they originated from. So, when loaded, assets without a chain
+		// ID are automatically considered to originate from Ethereum Mainnet.
+		if erc20 := p.Details.GetERC20(); erc20 != nil && erc20.ChainID == "" {
+			erc20.ChainID = s.ethClient.ConfiguredChainID()
+		}
+	}
+}
```

Comment:
We will need to send an event to the datanode as well here to propogate this change.

Reply:
Right !